Task4(homework4)

iryna-pantsialeyeva opened the pull request:
I kindly ask you to review my code.
Please don't merge this pull request.

iryna-pantsialeyeva committed:
added SecondsCounter (task from page 12)

iryna-pantsialeyeva committed:
added ReverseNumberWriter (task from page 12)

iryna-pantsialeyeva committed:
added DigitDifferencesFinder (task from page 11)

iryna-pantsialeyeva committed:
add homework4 (from Java Beginner Book [p.19] and Excel file)

i-velichko commented on src/task4/Invitation.java:
```diff
@@ -0,0 +1,37 @@
+
+        System.out.print("Enter your name: ");
+        String name = sc.nextLine();
+        System.out.print("Enter your sex (male of female): ");
```
In this implementation I don't see the reason of this question. The choice does not affect further answers in any way.

RomanKashenok replied:
так а на строке 22?

iryna-pantsialeyeva replied:
The choise of sort of drinks depends on the value of variable "sex".

RomanKashenok commented on src/task3/SecondsCounter.java:
```diff
@@ -0,0 +1,28 @@
+        System.out.println(seconds / secondsInWeek + " + " + seconds % secondsInWeek / secondsInDay + " + " +
+                (seconds % secondsInWeek % secondsInDay) / secondsInHour + " + " + ((seconds % secondsInWeek %
+                secondsInDay) % secondsInHour) / secondsInMinute + " + " + ((seconds % secondsInWeek %
+                secondsInDay) % secondsInHour) % secondsInMinute);
```
переиспользуй свои вычисления. И сделай вывод более понятным (хотя бы как в примере)

iryna-pantsialeyeva replied:
Прошу не оценивать таски из package Task3, вложенные в branch Task4. Причина - в комментарии к классу DigitDifferencesFinder.

iryna-pantsialeyeva replied:
По договоренности подтянула сюда решение из task3.

RomanKashenok commented on src/task3/ReverseNumberWriter.java:
```diff
@@ -0,0 +1,22 @@
+            // получаем СТРОКУ в виде числа путем перестановки (например, для 1000 получим 0001)
+            System.out.println("" + number % 10 + number % 100 / 10 + (number / 100) % 10 + number / 1000);
+            // получаем ЧИСЛО, полученное выписыванием в обратном порядке
+            System.out.println(number % 10 * 1000 + number % 100 / 10 * 100 + (number / 100) % 10 * 10 + number / 1000);
```
это я уже где-то комментировал

iryna-pantsialeyeva replied:
Прошу не оценивать таски из package Task3, вложенные в branch Task4. Причина - в комментарии к классу DigitDifferencesFinder.

iryna-pantsialeyeva replied:
По договоренности подтянула сюда решение из task3.

RomanKashenok commented on src/task3/DigitDifferencesFinder.java:
```diff
@@ -0,0 +1,29 @@
+        }
+        if (!equivalenceIndicator) {
+            System.out.println("Введенное число не содержит одинаковые цифры");
+        }
```
где-то уже видел

iryna-pantsialeyeva replied:
Сорри. Неправильно создавался branch (не в main), потому подтянулись классы из package Task3. 
Прошу не обращать на них внимание. В branch task3 все комменты учла и пофиксила.
Последние таски создавались правильно - из мастера.

iryna-pantsialeyeva replied:
По договоренности подтянула сюда решение из task3.

RomanKashenok commented on src/task4/FiveWayToCreateString.java:
```diff
@@ -0,0 +1,19 @@
+        // 3rd way to create a string
+        String string3 = string1 + string2;
+
+        StringBuilder string4 = new StringBuilder("4th way to create a string");
```
это не строка

iryna-pantsialeyeva replied:
Done.
Добавила еще 1 способ, т.к. StringBuilder сам по себе строку не создает (а создает экземпляр своего класса).

RomanKashenok commented on src/task4/FiveWayToCreateString.java:
```diff
@@ -0,0 +1,19 @@
+
+        StringBuilder string4 = new StringBuilder("4th way to create a string");
+
+        String string5 = string4.replace(0, 2, "5th").toString();
```
ладно )

RomanKashenok commented on src/task4/MonthsAndSeason.java:
```diff
@@ -0,0 +1,67 @@
+        }
+
+        switch (month) {
+            case "January":
```
давай объединим case с совпадающими выводами

iryna-pantsialeyeva replied:
Done.

iryna-pantsialeyeva committed:
add the update of  MonthsAndSeason(task4) by request in PR
cases with the same outputs are combined

iryna-pantsialeyeva committed:
add the update of FiveWayToCreateString(task4) by request in PR
additional way of string creation are added

iryna-pantsialeyeva committed:
in branch task4 add the update of the following tasks by request in PR:
           - DigitDifferencesFinder(task3)
           - ReverseNumberWriter(task3)
           - SecondsCounter(task3)
